api/src/main/java/org/openmrs/api/impl/PatientServiceImpl.java

```diff
@@ -117,6 +117,29 @@ public void onShutdown() {
 	 */
 	@Override
 	public Patient savePatient(Patient patient) throws APIException {
+		requireAppropriatePatientModificationPrivilege(patient);
+
+		if (!patient.getVoided() && patient.getIdentifiers().size() == 1) {
+			patient.getPatientIdentifier().setPreferred(true);
+		}
+
+		if (!patient.getVoided()) {
+			checkPatientIdentifiers(patient);
+		}
+
+		setPreferredPatientIdentifier(patient);
+		setPreferredPatientName(patient);
+		setPreferredPatientAddress(patient);
+
+		return dao.savePatient(patient);
+	}
+
+	/**
+	 * Checks if user has the required privileges to access patient details
+	 *
+	 * @param patient Patient whose details are to be saved
```

Comment:
@MalshaL please remove all the javadocs you added. Note how you just repeat what the method name + parameter already say? This just takes up space and the readers time without adding more information.

Of course if you would write a public method it should have javadocs.